### PR TITLE
Fix not supporting cors when options not defined

### PR DESCRIPTION
**Pull request by binodmx**
## Purpose
$subject

### CODE REVIEW

File: gateway/enforcer/org.wso2.apk.enforcer/src/main/java/org/wso2/apk/enforcer/cors/CorsFilter.java [hunk 1]
```diff
@@ -62,33 +62,18 @@ public boolean handleRequest(RequestContext requestContext) {
...
-                    for (ResourceConfig resourceConfig : requestContext.getMatchedAPI().getResources()) {
-                        if (!resourceConfig.getPath().equals(requestContext.getRequestPathTemplate())) {
-                            continue;
-                        }
-                        allowedMethodsBuilder.append(", ").append(resourceConfig.getMethod().name());
-                    }
-                }
-                requestContext.getProperties()
-                        .put(APIConstants.MessageFormat.STATUS_CODE, HttpConstants.NO_CONTENT_STATUS_CODE);
-                requestContext.addOrModifyHeaders(HttpConstants.ALLOW_HEADER, allowedMethodsBuilder.toString());
```
Comment (AmaliMatharaarachchi) [inline gateway/enforcer/org.wso2.apk.enforcer/src/main/java/org/wso2/apk/enforcer/cors/CorsFilter.java:80]: why did you remove it?

Comment (binodmx) [reply gateway/enforcer/org.wso2.apk.enforcer/src/main/java/org/wso2/apk/enforcer/cors/CorsFilter.java]: Resolved by https://github.com/wso2/apk/pull/1236/commits/d507691ca795a4060ed5fc771dac025b299d23a2

File: gateway/enforcer/org.wso2.apk.enforcer/src/main/java/org/wso2/apk/enforcer/cors/CorsFilter.java [hunk 2]
```diff
@@ -62,33 +62,18 @@ public boolean handleRequest(RequestContext requestContext) {
...
                 logger.debug("OPTIONS request received for " +
                         requestContext.getMatchedAPI().getResources().get(0).getPath() +
-                        ". Responded with allow header : " + allowedMethodsBuilder.toString());
+                        ". Responded with 403 status code.");
```
[the remark this answers is not in the document]
Comment (binodmx) [reply gateway/enforcer/org.wso2.apk.enforcer/src/main/java/org/wso2/apk/enforcer/cors/CorsFilter.java]: Resolved by https://github.com/wso2/apk/pull/1236/commits/d507691ca795a4060ed5fc771dac025b299d23a2

**Review by AmaliMatharaarachchi — APPROVED** [review]
